### PR TITLE
Refactor and bugfixing/add edit product

### PR DESCRIPTION
**Pull request by anetta999**
## Describe your changes

- added method setNewColors at addProductSlice to save in store changed colors after editing the product
- added method at addProductSlice to clear redux persist after logout action
- added brands request at VendorProductService to get the list of all brands from server instead of taking it from constant
- created useGetProductBrands hook to store the server request logic and make it reusable
- replaced colors select with multi-select at ProductDescriptionForm at VendorAddProduct/VendorEditProduct pages
- created reusable RejectProductFlowModal component
- added reject adding product modal (RejectProductFlowModal) at VendorAddProductPage
- replaced RejectEditingModal with RejectProductFlowModal at VendorEditProductPage
- enhanced pdf upload button logic at ProductDescriptionForm at VendorAddProduct/VendorEditProduct pages
- fixed bugs regarding the active state of "Next" button at ProductDescriptionForm at VendorAddProduct/VendorEditProduct pages
- changed navigation from orders to product list page after completing adding/editing product flow
- fixed sizes constants

## Issue ticket code (and/or) and link

- [Link to JIRA ticket](https://2024internship.atlassian.net/browse/SCRUM-83, https://2024internship.atlassian.net/browse/SCRUM-85, https://2024internship.atlassian.net/browse/SCRUM-86)

### **General**

- [x] Assigned myself to the PR
- [x] Assigned the appropriate labels to the PR
- [x] Assigned the appropriate reviewers to the PR
- [ ] Updated the documentation
- [x] Performed a self-review of my code
- [x] Types for input and output parameters
- [x] Don't have "any" on my code
- [x] Used the try/catch pattern for error handling
- [x] Don't have magic numbers
- [x] Compare only with constants not with strings
- [x] No ternary operator inside the ternary operator
- [x] Don't have commented code
- [x] no links in the code, env links should be in env file (for example: server url), constant links (for example default avatar URL) should be in constant file.
- [x] Used camelCase for variables and functions
- [ ] Date and time formats are on the constants
- [ ] Functions are public only if it's used outside the class
- [x] No hardcoded values
- [ ] covered by tests
- [x] Check your commit messages meet the [conventional commit format](https://www.conventionalcommits.org/en/v1.0.0/).

### Frontend

- [x] Components and business logic are separated
- [x] Colors, Font Size, and Font Name is on the theme or in the constants
- [x] No text in the components, use i18n approach
- [x] No inline styles
- [x] Imports are absolute
- [x] Attach a screenshot if PR has visual changes.

Снимок экрана 2024-08-10 в 18 31 53
Снимок экрана 2024-08-10 в 18 33 03
Снимок экрана 2024-08-10 в 18 34 34

### CODE REVIEW

File: src/pages/vendor/VendorEditProductPage/FinishForm/index.tsx [hunk 1]
```diff
@@ -104,7 +103,8 @@ function FinishForm() {
       dispatch(setPrice(parseFloat(values.price)));
       dispatch(resetAddProduct());
       showToast('success', t('editProduct.editingSuccess'));
-      navigate(`${urls.VENDOR}/${urls.VENDOR_ORDERS}`);
+      // navigate(`${urls.VENDOR}/${urls.VENDOR_ORDERS}`);
+      navigate(urls.VENDOR_GLOBAL_PRODUCTS);
```
Comment (likeri29) [inline src/pages/vendor/VendorEditProductPage/FinishForm/index.tsx:106]: delete comment

File: src/pages/vendor/VendorEditProductPage/ProductDescriptionForm/index.tsx [hunk 2]
```diff
@@ -306,9 +316,19 @@ function ProductDescriptionForm() {
...
               displayEmpty
-              value={productColor}
-              onChange={(v) => setProductColor(String(v.target.value))}
+              value={productColors}
+              renderValue={(selected) => {
+                const selectedColors = selected as colorsOptions[];
+
+                if (selectedColors.length === 0) {
+                  return <p>{selectedProductColors.join(' ')}</p>;
+                }
```
Comment (likeri29) [inline src/pages/vendor/VendorEditProductPage/ProductDescriptionForm/index.tsx:329]: use typography

File: src/redux/addProduct/addProductSlice.ts [hunk 3]
```diff
@@ -65,9 +67,13 @@ export const addProductSlice = createSlice({
     setSize(state, action: PayloadAction<string>) {
       state.size = action.payload;
     },
-    setColor(state, action: PayloadAction<string>) {
+    setColors(state, action: PayloadAction<string[]>) {
+      // state.colors = [];
+      state.colors = action.payload;
```
Comment (likeri29) [inline src/redux/addProduct/addProductSlice.ts:71]: delete comment

**Review by likeri29 — APPROVED** [review]

File: src/pages/vendor/VendorEditProductPage/FinishForm/index.tsx (hunk 1 above)
Comment (anetta999) [reply src/pages/vendor/VendorEditProductPage/FinishForm/index.tsx:106]: fixed

File: src/pages/vendor/VendorEditProductPage/ProductDescriptionForm/index.tsx (hunk 2 above)
Comment (anetta999) [reply src/pages/vendor/VendorEditProductPage/ProductDescriptionForm/index.tsx:329]: fixed

File: src/redux/addProduct/addProductSlice.ts (hunk 3 above)
Comment (anetta999) [reply src/redux/addProduct/addProductSlice.ts:71]: fixed

**Pull request merged**